src/main/kotlin/com/sourcegraph/cody/chat/AgentChatSession.kt

```diff
@@ -58,11 +62,11 @@ private constructor(
             }
 
     val selectedItem =
-        chatPanel.llmDropdown.selectedItem?.let { ChatModel.fromDisplayNameNullable(it.toString()) }
+        chatPanel.llmDropdown.selectedItem?.let { (it as? ChatModelsResponse.ChatModelProvider) }
     val restoreParams =
         ChatRestoreParams(
             // TODO: Change in the agent handling chat restore with null model
-            model?.agentName ?: selectedItem?.agentName ?: DEFAULT_MODEL.agentName,
+            chatModelProvider?.model ?: selectedItem?.model ?: "", // fixme: empty string
```

Comment:
fix?

Reply:
let's just pass `null` here. it's not worse but the code is simpler